app/controllers/users_controller.rb

```diff
@@ -0,0 +1,18 @@
+class UsersController < ApplicationController
+  def index
+    # it'll show all users
+  end
+
+  def show
+    # it'll show users
+  end
+
+  def set_users
+    @set_users ||= User.includes(:posts).all
+  end
+
+  def set_user
+    @set_user ||= User.includes(:posts).find(params[:id])
+    @recent_posts = @set_user.recent_posts
+  end
+end
```

Comment:
[OPTIONAL] I can see you calling the set_user and set_users methods anywhere in the users_controller. In scenarios like this, it is a good practice to make the methods private and use tools like before_action to call the methods.